zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Page.java

```diff
@@ -0,0 +1,212 @@
+        pageCreationHook = new PageTypeUpdateHook(creationHooks);
+    }
+
+    private Map<PageType, PageUpdateHook> initialisePageDeletionHooks(DatasetAPIClient datasetAPIClient) {
+        APIDatasetLandingPageDeletionHook datasetLandingPageDeletionHook =
+                new APIDatasetLandingPageDeletionHook(datasetAPIClient);
+
+        Map<PageType, PageUpdateHook> deletionHooks = new HashMap<>();
+        deletionHooks.put(PageType.api_dataset_landing_page, datasetLandingPageDeletionHook);
+
+        return deletionHooks;
+    }
+
+    private Map<PageType, PageUpdateHook> initialisePageCreationHooks(DatasetAPIClient datasetAPIClient) {
+        APIDatasetLandingPageCreationHook datasetLandingPageCreationHook =
+                new APIDatasetLandingPageCreationHook(datasetAPIClient);
+
+        Map<PageType, PageUpdateHook> creationHooks = new HashMap<>();
+        creationHooks.put(PageType.api_dataset_landing_page, datasetLandingPageCreationHook);
+
+        return creationHooks;
+    }
+
+    /**
+     * Constructor allowing dependencies to be injected.
+     *
+     * @param zebedeeCmsService
+     * @param pageCreationHook
+     * @param pageDeletionHook
+     */
+    public Page(ZebedeeCmsService zebedeeCmsService, PageUpdateHook pageCreationHook, PageUpdateHook pageDeletionHook) {
+        this.zebedeeCmsService = zebedeeCmsService;
+        this.pageCreationHook = pageCreationHook;
+        this.pageDeletionHook = pageDeletionHook;
+    }
+
+    /**
+     * create a new page from the endpoint <code>/Content/[CollectionName]/?uri=[uri]</code>
+     *
+     * @param request  This should contain a X-Florence-Token header for the current session
+     * @param response Returns true or false according to whether the URI was deleted.
+     * @throws IOException           If an error occurs in processing data, typically to the filesystem, but also on the HTTP connection.
+     * @throws BadRequestException   If the request cannot be completed because of a problem with request parameters
+     * @throws NotFoundException     If the requested URI does not exist in the collection.
+     * @throws UnauthorizedException If the user does not have publisher permission.
+     * @throws ConflictException     If the URI is being edited in another collection
+     */
+    @POST
+    public void createPage(HttpServletRequest request, HttpServletResponse response) throws IOException, ZebedeeException, FileUploadException {
+
+        // We have to get the request InputStream before reading any request parameters
+        // otherwise the call to get a request parameter will actually consume the body:
+        try (InputStream requestBody = request.getInputStream()) {
+
+            Session session = zebedeeCmsService.getSession(request);
+            com.github.onsdigital.zebedee.model.Collection collection = zebedeeCmsService.getCollection(request);
+            String uri = request.getParameter("uri");
+            uri = trimZebedeeFileSuffix(uri);
```

Comment:
can you check if uri is null / empty and throw the appropriate exception/ logging etc

Reply:
done